### PR TITLE
wifi-util: do not ignore wifi iftype when the interface does not have SSID

### LINKED ISSUE #18059: ath10k radio driver when connected as a mesh-point it does not match WLANInterfaceType
**systemd version the issue has been seen with**
 > Seen on the following systemd versions 246.6 247.1 and master 

**Used distribution**
 > Gentoo

**Linux kernel version used** (`uname -a`)
 > 4.19.160-gentoo

**CPU architecture issue was seen on**
 > AMD G-T40E 

**Expected behaviour you didn't see**
 > When ath10k radio is connected as a mesh point.  I expect it to match my .network file with WLANInterfaceType="mesh-point"  

**Unexpected behaviour you saw**
 > It does not match.

**Steps to reproduce the problem**
```
  [Match]
  Name=wlp6s0
  WLANInterfaceType="mesh-point"
```

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #18059.

### CODE REVIEW

File: src/shared/wifi-util.c [hunk 1]
```diff
@@ -35,8 +29,10 @@ int wifi_get_interface(sd_netlink *genl, int ifindex, enum nl80211_iftype *iftyp
         }
         if (r < 0)
                 return log_debug_errno(r, "Failed to request information about wifi interface %d: %m", ifindex);
-        if (!reply)
+        if (!reply) {
+                log_debug_errno(r, "No reply is obtained for requesting information about wifi interface %d, ignoring.", ifindex);
```
Comment (bluca) [inline src/shared/wifi-util.c]: nit: "No reply received to request for information"

File: src/shared/wifi-util.c [hunk 2]
```diff
@@ -101,8 +97,10 @@ int wifi_get_station(sd_netlink *genl, int ifindex, struct ether_addr *bssid) {
         r = sd_netlink_call(genl, m, 0, &reply);
         if (r < 0)
                 return log_debug_errno(r, "Failed to request information about wifi station: %m");
-        if (!reply)
+        if (!reply) {
+                log_debug_errno(r, "No reply is obtained for requesting information about wifi station, ignoring.");
```
Comment (bluca) [inline src/shared/wifi-util.c]: same as above

**Comment by yuwata** [comment]
@bluca Thank you for the comments. Updated. PTAL.

**Review by bluca — APPROVED** [review]
lgtm

**Pull request merged**